src/thermo/Phase.cpp

```diff
@@ -172,20 +173,49 @@ void Phase::getAtoms(size_t k, double* atomArray) const
     }
 }
 
+size_t Phase::findSpeciesLower(const std::string& name) const
+{
+    size_t loc = npos;
+    std::string nLower = toLowerCopy(name);
+    for (const auto& k : m_speciesIndices) {
+        if (toLowerCopy(k.first) == nLower) {
+            if (loc == npos) {
+                loc = k.second;
+            } else {
+                throw CanteraError("Phase::findSpeciesLower",
+                                   "Lowercase species name '{}' is not unique. "
+                                   "Set Phase::caseSensitiveSpecies to true to "
+                                   "enforce case sensitive species names", nLower);
+            }
+        }
+    }
+    return loc;
+}
+
 size_t Phase::speciesIndex(const std::string& nameStr) const
 {
-    size_t loc = getValue(m_speciesIndices, toLowerCopy(nameStr), npos);
+    size_t loc = npos;
+    std::map<std::string, size_t>::const_iterator it;
+
+    it = m_speciesIndices.find(nameStr);
+    if (it != m_speciesIndices.end()) {
```

Comment:
I think this is a good place to use `auto` instead of writing out the complex iterator type.

Reply:
:sparkles: ... didn't know that one.